device/identity_data.go

```diff
@@ -69,14 +69,16 @@ func (id IdentityDataRunner) Get() (string, error) {
 	}
 
 	p := utils.KeyValParser{}
-	if err := p.Parse(out); err != nil {
-		return "", errors.Wrapf(err, "failed to parse identity data")
-	}
+	parseErr := p.Parse(out)
 
 	if err := cmd.Wait(); err != nil {
 		return "", errors.Wrapf(err, "wait for helper failed")
```

Comment:
super nit: technically here the wait error might be caused by the not-nil `parseErr`, so we could add parseErr.Error() if parseErr not nil to the error message.

Reply:
btw, not waiting for children is a common mistake, and we should have an explicit test for this scenario, as it can fill up the process table easily -- we have to be sure it never comes back.

Reply:
> super nit: technically here the wait error might be caused by the not-nil `parseErr`, so we could add parseErr.Error() if parseErr not nil to the error message.

Yeah, I tried `Wrap` with a nil error, but this panics, so we need extra logic then. But you're right, the error from `parseErr` can be relevant.

> btw, not waiting for children is a common mistake, and we should have an explicit test for this scenario, as it can fill up the process table easily -- we have to be sure it never comes back.

Alright, I'll try to add it.

Originally I thought that the garbage collection would eventually run `Wait` for us, which would mean that any zombie accumulation would be temporary. But it doesn't seem to have a finalizer for this, so this is indeed important.